src/cython/sktda/vector_methods.py

```diff
@@ -0,0 +1,404 @@
+
+        for i in range(num_diag):
+
+            diagram, num_pts_in_diag = X[i], X[i].shape[0]
+
+            ls = np.zeros([self.num_landscapes, self.resolution])
+
+            events = []
+            for j in range(self.resolution):
+                events.append([])
+
+            for j in range(num_pts_in_diag):
+                [px,py] = diagram[j,:]
+                min_idx = np.minimum(np.maximum(np.ceil((px          - self.ls_range[0]) / step_x).astype(int), 0), self.resolution)
+                mid_idx = np.minimum(np.maximum(np.ceil((0.5*(py+px) - self.ls_range[0]) / step_x).astype(int), 0), self.resolution)
+                max_idx = np.minimum(np.maximum(np.ceil((py          - self.ls_range[0]) / step_x).astype(int), 0), self.resolution)
+
+                if min_idx < self.resolution and max_idx > 0:
+
+                    landscape_value = self.ls_range[0] + min_idx * step_x - px
+                    for k in range(min_idx, mid_idx):
+                        events[k].append(landscape_value)
+                        landscape_value += step_x
+
+                    landscape_value = py - self.ls_range[0] - mid_idx * step_x
+                    for k in range(mid_idx, max_idx):
+                        events[k].append(landscape_value)
+                        landscape_value -= step_x
+
+            for j in range(self.resolution):
+                events[j].sort(reverse=True)
+                for k in range( min(self.num_landscapes, len(events[j])) ):
+                    ls[k,j] = events[j][k]
+
+            Xfit.append(np.sqrt(2)*np.reshape(ls,[1,-1]))
+            Xfit = np.concatenate(Xfit,0)
+
+        return Xfit
+
+class Silhouette(BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing persistence silhouettes from a list of persistence diagrams. A persistence silhouette is computed by taking a weighted average of the collection of 1D piecewise-linear functions given by the persistence landscapes, and then by uniformly sampling this average on a given range. Finally, the corresponding vector of samples is returned. See https://arxiv.org/abs/1312.0308 for more details.
+    """
+    def __init__(self, weight=lambda x: 1, resolution=100, sh_range=[np.nan, np.nan]):
+        """
+        Constructor for the Silhouette class.
+
+        Attributes:
+            weight (function): weight function for the persistence diagram points (default constant function, ie lambda x: 1). This function must be defined on 2D points, ie on lists or numpy arrays of the form [p_x,p_y].
+            resolution (int): number of samples for the weighted average (default 100).
+            sh_range ([double, double]): minimum and maximum for the weighted average domain, of the form [x_min, x_max] (default [numpy.nan, numpy.nan]). It is the interval on which samples will be drawn uniformly. If one of the values is numpy.nan, it can be computed from the persistence diagrams with the fit() method.
```

Comment:
Could we give this parameter the same name in landscapes and silhouettes? Maybe just `range`, or something more explicit.

Reply:
I renamed it to `sample_range` for `Landscape`, `Silhouette`, `BettiCurve` and `Entropy` classes